e2e_playwright/st_dialog_nested_test.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+
+def test_nested_dialogs(app: Page):
+    """Test that st.dialog may not be nested inside other dialogs."""
+    exception_message = app.locator(".stException .message")
+
+    expect(exception_message).to_have_text(
+        "StreamlitAPIException: Dialogs may not be nested inside other dialogs."
+    )
```

Comment:
nit: I think it would be slightly better not to rely on classes for testing, something like this should also work:

```suggestion
    """Test that st.dialog may not be nested inside other dialogs."""
    exception_message = app.get_by_test_id("stException")

    expect(exception_message).to_contain_text(
        "StreamlitAPIException: Dialogs may not be nested inside other dialogs."
    )
```